ebos/eclproblem.hh

```diff
@@ -1443,6 +1480,38 @@ public:
         }
 
         aquiferModel_.addToSource(rate, context, spaceIdx, timeIdx);
+
+        // if requested, compensate systematic mass loss for cells which were "well
+        // behaved" in the last time step
+        if (enableDriftCompensation_) {
+            unsigned globalDofIdx = context.globalSpaceIndex(spaceIdx, timeIdx);
+            const auto& intQuants = context.intensiveQuantities(spaceIdx, timeIdx);
+            const auto& simulator = this->simulator();
+            const auto& model = this->model();
+
+            // we need a higher maxCompensation than the Newton tolerance because the
+            // current time step might be shorter than the last one
+            Scalar maxCompensation = 10.0*model.newtonMethod().tolerance();
+
+            Scalar poro = intQuants.referencePorosity();
+            Scalar dt = simulator.timeStepSize();
+
+            EqVector dofDriftRate = drift_[globalDofIdx];
+            dofDriftRate /= dt*context.dofTotalVolume(spaceIdx, timeIdx);
+
+            // compute the weighted total drift rate
+            Scalar totalDriftRate = 0.0;
+            for (unsigned eqIdx = 0; eqIdx < numEq; ++ eqIdx)
+                totalDriftRate +=
+                    std::abs(dofDriftRate[eqIdx])*dt*model.eqWeight(globalDofIdx, eqIdx)/poro;
+
+            // make sure that we do not exceed the maximum rate of drift compensation
+            if (totalDriftRate > maxCompensation)
+                dofDriftRate *= maxCompensation/totalDriftRate;
+
```

Comment:
The implementation here is slightly different from the earlier prototype.  How well did you test this? How sensitive is the results on the particular choice of maxCompensation?

Reply:
I ran it on Norne and M2R5, and it seemed to work well. more extensive testing is definitely in order, though...

Reply:
Ok. I will do some more testing. Since it is within the experimental mode I will merge it as is.